test/loopback-dl.test.js

```diff
@@ -837,6 +837,78 @@ describe('Models attached to a dataSource', function() {
 
   });
 
+  context('replaceOrCreate', function() {
+    it('should replace the instance', function(done) {
+      Post.create({title: 'a', content: 'AAA'}, function(err, post) {
+        if (err) return done(err);
+        post = post.toObject();
+        delete post.content;
+        Post.replaceOrCreate(post, function(err, p) {
+          if (err) return done(err);
+          p.id.should.equal(post.id);
+          p.title.should.equal('a'); 
+          should.not.exist(p.content);
+          should.not.exist(p._id);
+          Post.findById(post.id, function(err, p) {
+            if (err) return done(err);
+            p.id.should.equal(post.id);
+            p.title.should.equal('a'); 
+            should.not.exist(post.content);
+            should.not.exist(p._id);
+            done();
+          });
+        });
+      });
+    });
+
+    it('should replace with new data', function(done) {
```

Comment:
What's the difference between this test and the test above? It seems to me this second one is a super-set of the first one, and thus it's enough to keep only this second one.